pylint/test/functional/consider_using_in.py

```diff
@@ -0,0 +1,41 @@
+# pylint: disable=missing-docstring, invalid-name, pointless-statement, misplaced-comparison-constant, undefined-variable, literal-comparison, line-too-long, unneeded-not
+
+value = value1 = 1
+value2 = 2
+a_set = {1, 2, 3}
+a_list = [1, 2, 3]
+a_str = '1'
+
+# Positive
+value == 1 or value == 1  # [consider-using-in]
+value == 1 or value == 2  # [consider-using-in]
+'value' == value or 'value' == value  # [consider-using-in]
+value == 1 or value == undef_value  # [consider-using-in]
+value == 1 or value == 2 or value == 3  # [consider-using-in]
+value == '2' or value == 1  # [consider-using-in]
+1 == value or 2 == value  # [consider-using-in]
+1 == value or value == 2  # [consider-using-in]
+value == 1 or value == a_list  # [consider-using-in]
+value == a_set or value == a_list or value == a_str  # [consider-using-in]
+value != 1 and value != 2  # [consider-using-in]
+value1 == value2 or value2 == value1  # [consider-using-in]
+a_list == [1, 2, 3] or a_list == []  # [consider-using-in]
+
+# Undecided
+'value' == 1 or 'value' == 2  # no variables
```

Comment:
Not sure if this should be the responsibility of this checker, I think we should just stick to variables for now.